concrete/src/Package/Package.php

```diff
@@ -609,9 +615,13 @@ public function restore()
         return false;
     }
 
+    public function getEntityManagerProvider()
```

Comment:
I'd like to see this as part of an optional generic interface instead of a new public function.

To use the method, you can do:
```php
if ($package instanceof \Concrete\Core\Entity\Manager\ProviderInterface) {
    $provider = $package;
} elseif ($package instanceof \Concrete\Core\Entity\Manager\ProviderAggregateInterface) {
    $provider = $package->getEntityManagerProvider();
} else {
    $provider = new DefaultPackageProvider($package);
}
```

then if someone needs to override the default entity manager, they can do so by implementing the interface. And we can potentially reuse this interface in multiple places without having to implement multiple copies of the function. 
This helps prevent our classes from ballooning into having dozens of responsibilities since each responsibility is defined as a separate type. Even though implementations may use each of these responsibilities, at least the core remains clean.
It also helps us when it comes time to remove certain api's. Since there's no parent code, we can just deprecate the interface and leave it at that, no need for backwards incompatible changes.